[Pangle]Support adaptive banner and add and remove some APIs

PangleZhangketong opened the pull request:
1. Added adaptive banner style
2. Added API to PAGConfig.Builder
`PAGConfig.Builder().setAdxId(String adxId)`
3. Added method to PAGSdk
`PAGSdk.getBiddingToken(context,biddingRequest,biddingTokenCallback)`

PangleZhangketong commented on ThirdPartyAdapters/pangle/pangle/src/main/java/com/google/ads/mediation/pangle/PangleMediationAdapter.java:
```diff
@@ -300,34 +300,36 @@ public static int getGDPRConsent() {
    */
-  public static void setDoNotSell(@PAGDoNotSellType int ccpa) {
-    setDoNotSell(ccpa, new PangleSdkWrapper());
+  public static void setPAConsent(@PAGConstant.PAGPAConsentType int pa) {
```
CCPA：V71 SDK upgrade removes the CCPA flag. The CCPA flag is replaced by the new PA consent field in the current version.
PA Consent：Pangle SDK interface has added 「PA consent」 field, requiring publishers to postback whether the user of the ad request accepts to be served with Personalized Ad. If publishers do not postback, the default is ON. We will refer to this value to serve ads.

thanvirm replied:
Thanks for the explanation!

PangleZhangketong commented on ThirdPartyAdapters/pangle/pangle/src/main/java/com/google/ads/mediation/pangle/PangleInitializer.java:
```diff
@@ -94,12 +94,12 @@ public void initialize(
         pangleFactory
             .createPAGConfigBuilder()
             .appId(appId)
-            .setChildDirected(PanglePrivacyConfig.getCoppa())
```
Removal of the COPPA flag is to help facilitate publishers' compliance with Pangle's terms. Under Pangle's terms, publishers are required not "to make available to Pangle, nor enable Pangle to collect, Sensitive Data or Personal Data relating to Children." If publishers know that an ad request is of a child as defined by the Pangle terms, publishers should not initialize the Pangle SDK for those users' ad requests.

thanvirm replied:
Thanks for the explanation.

Could you update your PR to include if-blocks that check whether user is a child (i.e. whether Google's TFCD/TFUA bit is true or not) and blocks init, signal collection and ad load if the user is a child? This would be similar to what is done in the AppLovin adapter: https://github.com/googleads/googleads-mobile-android-mediation/commit/d6c087f33243178c08615ca3c636963ed64f6c0d

Without that check, if Pangle is added as an ad source on Admob FE by the publisher, Pangle SDK will get initialized, signal will be collected and load-ad requests will be sent to Pangle even if the user is a child.

Thanks!

PangleZhangketong replied:
Thanks for your suggestion, I have updated my code according to this logic

thanvirm replied:
Thank you for making the changes!

PangleZhangketong commented on ThirdPartyAdapters/pangle/pangle/build.gradle:
```diff
@@ -33,7 +33,7 @@ android {
 }
 
 dependencies {
-    implementation 'com.pangle.global:ads-sdk:6.5.0.6'
```
I have resynced with the main branch  and updated the Pangle adapter to depend on pag-sdk:7.1.0.5

thanvirm commented on ThirdPartyAdapters/pangle/pangle/src/main/java/com/google/ads/mediation/pangle/PangleMediationAdapter.java:
```diff
@@ -300,34 +332,36 @@ public static int getGDPRConsent() {
     }
     if (pangleSdkWrapper.isInitSuccess()) {
-      pangleSdkWrapper.setDoNotSell(ccpa);
+      pangleSdkWrapper.setPAConsent(pa);
```
In this if statement too, you want to check if the user is a child?

PangleZhangketong replied:
This API does not need to check whether the user is a child. This API only has the ability to save the PA consent flag to memory and does not involve any data collection or transmission.

thanvirm reviewed (changes requested):
Just left one comment. Looks good otherwise. Thank you for making the changes!

thanvirm commented:
Hey, I have [published](https://github.com/googleads/googleads-mobile-android-mediation/commit/e8009e8e50749650f067c5cc99d1ed7247ab8da0) to GitHub our unit tests for the Pangle adapter. Those unit tests are unfortunately failing with your changes. Could you please fix the unit tests? Thank you so much!

PangleZhangketong commented:
> Hey, I have [published](https://github.com/googleads/googleads-mobile-android-mediation/commit/e8009e8e50749650f067c5cc99d1ed7247ab8da0) to GitHub our unit tests for the Pangle adapter. Those unit tests are unfortunately failing with your changes. Could you please fix the unit tests? Thank you so much!

PangleZhangketong commented:
@thanvirm  i fixed  the unit tests

PangleZhangketong commented:
> Hey, I have [published](https://github.com/googleads/googleads-mobile-android-mediation/commit/e8009e8e50749650f067c5cc99d1ed7247ab8da0) to GitHub our unit tests for the Pangle adapter. Those unit tests are unfortunately failing with your changes. Could you please fix the unit tests? Thank you so much!

i fixed the unit tests

PangleZhangketong commented:
> > > Hey, I have [published](https://github.com/googleads/googleads-mobile-android-mediation/commit/e8009e8e50749650f067c5cc99d1ed7247ab8da0) to GitHub our unit tests for the Pangle adapter. Those unit tests are unfortunately failing with your changes. Could you please fix the unit tests? Thank you so much!
> > 
> > 
> > i fixed the unit tests
> 
> Thank you for updating the unit tests.
> 
> Unfortunately, looks like a few of the unit tests are still failing (https://github.com/googleads/googleads-mobile-android-mediation/actions/runs/14898181965/job/41894096239?pr=567). Could you fix those? Thank you!

I updated the unit tests and successfully executed the cases locally.

guchan-fout commented:
@thanvirm Hi, it's Gu from Pangle team, hope you are doing well.
May I know if is there anything you need us to do? we completed the unit test and hope you can check

thanvirm commented:
> I updated the unit tests and successfully executed the cases locally.

Hey, the PR looks good to me. I need an internal approval in our internal codebase before this can be merged. I think I will get that approval soon. No action needed from you. Thanks for fixing the unit tests!

guchan-fout commented:
@thanvirm thx for the reply, may I ask is that the v71 already can be downloaded via gradle? Thank you
